### PR TITLE
[Bugfix] [zos_job_output] [zos_job_query] Fix username validator in job util

### LINKED ISSUE #1486: [Bug][zos_job_output] ValueError('Invalid argument \"@02858\" for type \"qualifier_pattern\".')
### Is there an existing issue for this?

- [X] There are no existing issues.

### Bug description

Tried running this task but returned an error for the owner.
owner: "@02858"

  - name: Get output
    zos_job_output:
      job_name: "*"
      job_id:   "J*"
      ddname:   "?"
      owner:    "{{ansible_user}}"

--

TASK [Get output] ***************************************************************************************************************
fatal: [mvs-sysd]: FAILED! => {"changed": false, "msg": "ValueError('Invalid argument \"@02858\" for type \"qualifier_pattern\".')"}

### IBM z/OS Ansible core Version

v1.9.0

### IBM Z Open Automation Utilities

v1.2.5 (default)

### IBM Enterprise Python

v3.11.x (default)

### ansible-version

v2.16.x (default)

### z/OS version

v2.5 (default)

### Ansible module

zos_job_output

### Playbook verbosity output.

Ran this task returning the output snippet.
  - name: Get output
    zos_job_output:
      job_id: "{{JOBID}}"
      ddname: "{{JOBLG}}"
    register: job_out2
--
                    "job_id": "J0020333",
                    "job_name": "@02858CZ",
                    "owner": "@02858",
--
Tried running this task but returned an error for the owner.
  - name: Get output
    zos_job_output:
      job_name: "*"
      job_id:   "J*"
      ddname:   "?"
      owner:    "{{ansible_user}}"

--

TASK [Get output] ***************************************************************************************************************
fatal: [mvs-sysd]: FAILED! => {"changed": false, "msg": "ValueError('Invalid argument \"@02858\" for type \"qualifier_pattern\".')"}

### Ansible configuration.

_No response_

### Contents of the inventory

_No response_

### Contents of `group_vars` or `host_vars`

_No response_

### PR DESCRIPTION
**Pull request by rexemin**
##### SUMMARY
Fixes #1486.

##### ISSUE TYPE
- Bugfix Pull Request

##### COMPONENT NAME
- module_utils/job.py

##### ADDITIONAL INFORMATION
Added a new validator function inside the job util. Can change the function from the util to the BetterArgParser file if needed. Didn't change the qualifier validator previously used since job IDs do not use @, #, or $.

### CODE REVIEW

**Comment by rexemin** [comment]
Screenshot 2024-08-20 at 13 30 29
Screenshot 2024-08-20 at 13 30 40

**Review by richp405 — APPROVED** [review]

File: changelogs/fragments/1661-job-owner-valid-characters.yml [hunk 1]
```diff
@@ -0,0 +1,9 @@
+bugfixes:
+  - zos_job_output - RACF user names containing a ``@`` raised an invalid
```
Comment (fernandofloresg) [inline changelogs/fragments/1661-job-owner-valid-characters.yml:2]: I think you can add another sample of failing character or just say it didn't work with other special characters.

`RACF user names containing a ``@`` or other special characters raised an invalid argument error.`

File: plugins/module_utils/job.py [hunk 2]
```diff
@@ -525,7 +526,44 @@ def _ddname_pattern(contents, resolve_dependencies):
...
         raise ValueError(
-            'Invalid argument type for "{0}". expected "ddname_pattern"'.format(
+            'Invalid argument type for "{0}". Expected "ddname_pattern"'.format(
+                contents
+            )
+        )
+    return str(contents)
+
+
+def _username_pattern(contents, resolve_dependencies):
```
Comment (fernandofloresg) [inline plugins/module_utils/job.py:536]: other type validations are in `plugins/module_utils/better_arg_parser.py` I think this should be included there too.

**Review by fernandofloresg — CHANGES_REQUESTED** [review]
Just requested a couple of changes. It seems correct to use a new type to validate username instead of modifying the job_id type.

**Review by fernandofloresg — APPROVED** [review]
lgtm

**Review by AndreMarcel99 — APPROVED** [review]

**Pull request merged**